src/directives/css.ts

```diff
@@ -0,0 +1,99 @@
+ * It takes the properties in the `classInfo` argument and applies each
+ * property name to the element's `classList` based on if the property value is truthy.
+ * For example `{foo: bar}` applies the class `foo` if the value of `bar` is truthy.
+ * @param classInfo {ClassInfo}
+ */
+export const classMap = (classInfo: ClassInfo): Directive<AttributePart> =>
+  directive((part: AttributePart) => {
+    if (!(part instanceof AttributePart) || (part instanceof PropertyPart) ||
+        part.committer.name !== 'class' || part.committer.parts.length > 1) {
+      throw new Error('The `classMap` directive must be used in the `class` attribute ' +
+          'and must be the only part in the attribute.');
+    }
+    // handle static classes
+    if (!classMapStatics.has(part)) {
+      part.committer.element.className =
+          part.committer.strings.reduce((a, c) => `${a} ${c}`);
+      classMapStatics.add(part);
+    }
+    // remove old classes that no longer apply
+    const oldInfo = classMapCache.get(part);
+    for (const name in oldInfo) {
+      if (!(name in classInfo)) {
+        part.committer.element.classList.remove(name);
+      }
+    }
+    // add new classes
+    for (const name in classInfo) {
+      if (!oldInfo || (oldInfo[name] !== classInfo[name])) {
+        // We explicitly want a loose truthy check here because
+        // it seems more convenient that '' and 0 are skipped.
+        part.committer.element.classList.toggle(name, Boolean(classInfo[name]));
+      }
+    }
+    classMapCache.set(part, classInfo);
+});
+
+const styleMapCache = new WeakMap();
+const styleMapStatics = new WeakSet();
+
+/**
+ * A directive that applies css properties. This must be used in the `style` attribute.
+ * It takes the property names in the `styleInfo` object and applies the property
+ * values as a css style propertes. For example `{backgroundColor: 'red', borderTop: '5px'}`
+ * sets these properties to the element's style.
+ * @param styleInfo {StyleInfo}
+ */
+export const styleMap = (styleInfo: StyleInfo): Directive<AttributePart> =>
+  directive((part: AttributePart) => {
+    if (!(part instanceof AttributePart) || (part instanceof PropertyPart) ||
+        part.committer.name !== 'style' || part.committer.parts.length > 1) {
+      throw new Error('The `styleMap` directive must be used in the style attribute ' +
+          'and must be the only part in the attribute.');
+    }
+    // handle static styles
+    if (!styleMapStatics.has(part)) {
+      (part.committer.element as HTMLElement).style.cssText =
+          part.committer.strings.reduce((a, c) => `${a} ${c}`);
+      styleMapStatics.add(part);
+    }
+    // remove old styles that no longer apply
+    const oldInfo = styleMapCache.get(part);
+    for (const name in oldInfo) {
+      if (!(name in styleInfo)) {
+        // setting to `null` to "unset" a previously applied value.
+        ((part.committer.element as HTMLElement).style as any)[name] = null;
```

Comment:
`element.style.foo = null` does not work in IE11 :( you have to use `foo.style.removeProperty(foo)`

Reply:
Seems to work in testing. Do you know if there are there certain properties for which this doesn't work? (`removeProperty` doesn't accept the camelCased property name, so would love to avoid that if possible).

Reply:
Try this jsbin: http://jsbin.com/kosikijule/edit?html,output the background color should be removed after 2 sec

I notice now that that it works for `height` but not for `backgroundColor`, but it can't be due to casing because I've had this issue a lot with properties like `overflow` when writing css animations.

I understand not wanting to use `removeProperty` as you'd need to convert camel to dash...

Reply:
You can reset by setting the CSS property to an empty string, that should work everywhere including IE11. e.g.
`element.style.foo = '';`